internal/clair/index.go

```diff
@@ -165,9 +174,30 @@ func (c *Client) submitManifest(renderManifest func() (Manifest, error)) (indexR
 		bytes.NewReader(jsonBytes),
 	)
 	if err != nil {
-		return indexReport{}, err
+		return indexReport{}, "", err
+	}
+
+	var indexReportResult indexReport
+	err = c.doRequest(req, &indexReportResult)
+	if err != nil {
+		return indexReport{}, "", err
 	}
-	var result indexReport
-	err = c.doRequest(req, &result)
-	return result, err
+
+	// get and return index state hash to later resubmit reports if the configuration changed
+	req, err = http.NewRequest(
+		http.MethodGet,
+		c.requestURL("indexer", "api", "v1", "index_state"),
+		bytes.NewReader(jsonBytes),
```

Comment:
GET requests don't accept bodies.
```suggestion
		nil,
```